recipe/multitenancy/recipe.go

```diff
@@ -68,13 +102,31 @@ func GetRecipeInstanceOrThrowError() (*Recipe, error) {
 	return nil, errors.New("Initialisation not done. Did you forget to call the init function?")
 }
 
+func GetRecipeInstance() *Recipe {
+	return singletonInstance
+}
+
 func recipeInit(config *multitenancymodels.TypeInput) supertokens.Recipe {
 	return func(appInfo supertokens.NormalisedAppinfo, onSuperTokensAPIError func(err error, req *http.Request, res http.ResponseWriter)) (*supertokens.RecipeModule, error) {
 		if singletonInstance == nil {
 			recipe, err := MakeRecipe(RECIPE_ID, appInfo, config, onSuperTokensAPIError)
 			if err != nil {
 				return nil, err
 			}
+
+			supertokens.AddPostInitCallback(func() error {
+				sessionRecipe, err := session.GetRecipeInstanceOrThrowError()
+
+				if err != nil {
+					return nil // skip adding claims if session recipe is not initialised
+				}
+
+				sessionRecipe.AddClaimFromOtherRecipe(multitenancyclaims.MultitenancyTenantIdClaim)
+				sessionRecipe.AddClaimFromOtherRecipe(multitenancyclaims.MultitenancyDomainsClaim)
```

Comment:
one issue with this is that even if someone is not going to use multi tenancy at all, their session will have that claims.. or if someone is going to use multi tenancy, but the user belongs to the default tenant, it will still be added to the session.

Reply:
Adding this only when the user has provided a function for `GetAllowedDomainsForTenantId`